### PR TITLE
Feature/add mac os example

### PR DESCRIPTION
**Pull request by andrew-winn-br**
Had to add Catalyst support because a Catalyst app uses the iOS/Catalyst version of a framework, not the macOS version.
This should complete #6 
Screen Shot 2022-02-03 at 11 54 36 AM
Screen Shot 2022-02-03 at 11 54 17 AM

### CODE REVIEW

**Comment by andrew-winn-br** [comment]
Linter error is:

**Comment by andrew-winn-br** [comment]
Linter error is:
```
PackageDescription.SupportedPlatform:5:24: note: 'macCatalyst' was introduced in PackageDescription 5.5
PackageDescription.SupportedPlatform:33:27: note: 'v13' was introduced in PackageDescription 5.5
```
and the current package description for Hyperspace is 5.1. I think I can just remove the .`macCatalyst(.v13)` line from Package.Swift and it should still be fine. `.macCatalyst` was added as a specific platform for SPM in Xcode 13, but Catalyst was supported before that.

**Comment by wmcginty** [comment]
@andrew-winn-br  I think we go ahead and upgrade the minimum tools version to 5.5, especially as this isn't going straight into a release we don't have to immediately worry about compatibility issues.

File: Example/Sources/macOS/ViewController.swift [hunk 1]
```diff
@@ -0,0 +1,96 @@
+//
+//  ViewController.swift
+//  macOS Example
+//
+//  Created by Andrew Winn on 2/2/22.
```
Comment (wmcginty) [inline Example/Sources/macOS/ViewController.swift:5]: Can you remove the `Created by` headers? Just so know one tries to track you down specifically months from now when the correct method would be creating an issue on the project.

**Review by wmcginty — CHANGES_REQUESTED** [review]
This mostly looks good to me, but can you also add a CHANGELOG.md entry?

File: Example/Sources/macOS/ViewController.swift (hunk 1 above)
Comment (andrew-winn-br) [reply Example/Sources/macOS/ViewController.swift:5]: Oops Yeah will do

Comment (andrew-winn-br) [reply Example/Sources/macOS/ViewController.swift]: done

**Review by wmcginty — APPROVED** [review]

File: Example/Sources/macOS/ViewController.swift [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                self?.presentAlert(titled: "Error", message: "\(error)")
+            }
+        }
+    }
+
+    private func deletePost(postId: Int) {
+        let deletePostRequest = Request<EmptyResponse, AnyError>.deletePost(withID: postId)
+
+        backendService.execute(request: deletePostRequest) { [weak self] result in
+            switch result {
```
Comment (br-earl-gaspard) [inline Example/Sources/macOS/ViewController.swift:71]: I noticed `debugPrint` statements for other `backendService.execute` calls but not one here.

**Review by br-earl-gaspard — CHANGES_REQUESTED** [review]

**Review by br-earl-gaspard — APPROVED** [review]

File: Example/Sources/macOS/ViewController.swift (hunk 2 above)
Comment (andrew-winn-br) [reply Example/Sources/macOS/ViewController.swift:73]: Done!

**Review by wmcginty — APPROVED** [review]

**Pull request merged**